mne_bids/tsv_handler.py

```diff
@@ -0,0 +1,147 @@
+import numpy as np
+from collections import OrderedDict, Counter
+
+
+def _from_tsv(fname, dtypes=None):
+    """Read a tsv file into an OrderedDict.
+
+    Parameters
+    ----------
+    fname : str
+        Path to the file being loaded.
+    dtypes : list
+        List of types to cast the values loaded as. This is specified column by
+        column.
+        Defaults to None. In this case all the data is loaded as strings.
+
+    Returns
+    -------
+    data : collections.OrderedDict
+        Keys are the column names, and values are the column data.
+    """
+    data = np.loadtxt(fname, dtype=str, delimiter='\t')
+    column_names = data[0, :]
+    info = data[1:, :]
+    data_dict = OrderedDict()
+    if isinstance(dtypes, (list, tuple)):
+        if len(dtypes) == info.shape[1]:
+            for i, name in enumerate(column_names):
+                data_dict[name] = info[:, i].astype(dtypes[i]).tolist()
+    else:
+        for i, name in enumerate(column_names):
+            data_dict[name] = info[:, i].tolist()
+    return data_dict
+
+
+def _to_tsv(data, fname):
+    """Write an OrderedDict into a tsv file.
+
+    Parameters
+    ----------
+    data : collections.OrderedDict
+        Ordered dictionary containing data to be written to a tsv file.
+    fname : str
+        Path to the file being written.
+    """
+    output = ''
+    n_rows = len(data[list(data.keys())[0]])
+    # write headings
+    output += '\t'.join(list(data.keys())) + '\n'
+
+    # write column data
+    for idx in range(n_rows):
+        row_data = list(str(data[key][idx]) for key in data)
+        output += '\t'.join(row_data)
+        if idx != n_rows - 1:
+            output += '\n'
+
+    with open(fname, 'w') as f:
+        f.write(output)
+
+
+def _combine(data1, data2, drop_column=None):
+    """Add two OrderedDict's together with optional dropping of repeated data.
+
+    Parameters
+    ----------
+    data1 : collections.OrderedDict
+        Original OrderedDict.
+    data2 : collections.OrderedDict
+        New OrderedDict to be added to the original.
+    drop_column : str
+        Name of the column to check for duplicate values in.
+        Any duplicates found will be dropped from the original data array (ie.
+        most recent value are kept).
+
+    Returns
+    -------
+    data : collections.OrderedDict
+        The new combined data.
+    """
+    new_data = data1.copy()
+    for key, value in data2.items():
+        new_data[key].extend(value)
+    if drop_column in new_data:
+        column_data = new_data[drop_column]
+        values_count = Counter(column_data)
+        for key, value in values_count.items():
+            # find the locations of the first n-1 values and remove the rows
+            for _ in range(value - 1):
+                idx = new_data[drop_column].index(key)
+                for column in new_data.values():
+                    del column[idx]
+    return new_data
+
+
+def _drop(data, values, column):
```

Comment:
do we need the `column` variable? I would just called the second argument `name` or something like that

Reply:
actually never mind my comment.